### ISSUE TITLE
tidyverse Error in date_names_lang(date_names) cannot open file sysdata.rdb No such file or directory

### ISSUE THREAD

**Issue by nickhepler** [tidyverse/tidyverse#46]
# Issue
I’m getting the following error attempting to use the read_csv from tidyverse: 
> Error in date_names_lang(date_names) : 
>   cannot open file '~/R/x86_64-pc-linux-gnu-library/3.3/readr/R/sysdata.rdb': No such file or directory

# Steps to Reproduce
1. Load tidyverse
`library("tidyverse", lib.loc="~/R/x86_64-pc-linux-gnu-library/3.3")`
2. Run read_csv command
`raw <- read_csv("https://data.ny.gov/api/views/q6ts-kwhk/rows.csv?accessType=DOWNLOAD")`
3. Result
> Error in date_names_lang(date_names) : 
>   cannot open file '~/R/x86_64-pc-linux-gnu-library/3.3/readr/R/sysdata.rdb': No such file or directory

# Steps Taken to Analyze Issue
1. Terminate R Session
2. Load readr
`library("readr", lib.loc="~/R/x86_64-pc-linux-gnu-library/3.3")`
3. Run read_csv command
`raw <- read_csv("https://data.ny.gov/api/views/q6ts-kwhk/rows.csv?accessType=DOWNLOAD")`
4. Result
> Parsed with column specification:
> cols(
>   Year = col_integer(),
>   `Incident Date` = col_character(),
>   `Incident Type` = col_character(),
>   Track = col_character(),
>   `Inv Location` = col_character(),
>   `Racing Type Description` = col_character(),
>   Division = col_character(),
>   `Weather Conditions` = col_character(),
>   Horse = col_character(),
>   Trainer = col_character(),
>   `Jockey Driver` = col_character(),
>   `Incident Description` = col_character(),
>   `Death or Injury` = col_character()
> )
5. Terminate R Session
6. Load readr
`library("readr", lib.loc="~/R/x86_64-pc-linux-gnu-library/3.3")`
7. Load tidyverse
`library("tidyverse", lib.loc="~/R/x86_64-pc-linux-gnu-library/3.3")`
8. Run read_csv command
`raw <- read_csv("https://data.ny.gov/api/views/q6ts-kwhk/rows.csv?accessType=DOWNLOAD")`
9. Result
> Parsed with column specification:
> cols(
>   Year = col_integer(),
>   `Incident Date` = col_character(),
>   `Incident Type` = col_character(),
>   Track = col_character(),
>   `Inv Location` = col_character(),
>   `Racing Type Description` = col_character(),
>   Division = col_character(),
>   `Weather Conditions` = col_character(),
>   Horse = col_character(),
>   Trainer = col_character(),
>   `Jockey Driver` = col_character(),
>   `Incident Description` = col_character(),
>   `Death or Injury` = col_character()
> )

# Steps Taken to Attempt Resolution
1. Remove readr package
`remove.packages("readr", lib="~/R/x86_64-pc-linux-gnu-library/3.3")`
2. Install readr package
`install.packages("readr")`
3. Remove tidyverse package
`remove.packages("tidyverse", lib="~/R/x86_64-pc-linux-gnu-library/3.3")`
4. Install tidyverse package
`install.packages("tidyverse")`

# Session Info
`devtools::session_info()`
> Session info -------------------------------------------------------------------------------------
>  setting  value                       
>  version  R version 3.3.3 (2017-03-06)
>  system   x86_64, linux-gnu           
>  ui       RStudio (1.0.136)           
>  language (EN)                        
>  collate  en_US.UTF-8                 
>  tz       America/New_York            
>  date     2017-03-07                  
> 
> Packages -----------------------------------------------------------------------------------------
>  package    * version date       source        
>  assertthat   0.1     2013-12-06 CRAN (R 3.3.2)
>  broom        0.4.2   2017-02-13 CRAN (R 3.3.2)
>  colorspace   1.3-2   2016-12-14 CRAN (R 3.3.2)
>  curl         2.3     2016-11-24 CRAN (R 3.3.2)
>  DBI          0.5-1   2016-09-10 CRAN (R 3.3.2)
>  devtools     1.12.0  2016-12-05 CRAN (R 3.3.3)
>  digest       0.6.12  2017-01-27 CRAN (R 3.3.2)
>  dplyr      * 0.5.0   2016-06-24 CRAN (R 3.3.2)
>  forcats      0.2.0   2017-01-23 CRAN (R 3.3.2)
>  foreign      0.8-67  2016-09-13 CRAN (R 3.3.1)
>  ggplot2    * 2.2.1   2016-12-30 CRAN (R 3.3.2)
>  gtable       0.2.0   2016-02-26 CRAN (R 3.3.2)
>  haven        1.0.0   2016-09-23 CRAN (R 3.3.2)
>  hms          0.3     2016-11-22 CRAN (R 3.3.2)
>  httr         1.2.1   2016-07-03 CRAN (R 3.3.2)
>  jsonlite     1.3     2017-02-28 CRAN (R 3.3.2)
>  lattice      0.20-34 2016-09-06 CRAN (R 3.3.1)
>  lazyeval     0.2.0   2016-06-12 CRAN (R 3.3.2)
>  lubridate    1.6.0   2016-09-13 CRAN (R 3.3.2)
>  magrittr     1.5     2014-11-22 CRAN (R 3.3.2)
>  memoise      1.0.0   2016-01-29 CRAN (R 3.3.3)
>  mnormt       1.5-5   2016-10-15 CRAN (R 3.3.2)
>  modelr       0.1.0   2016-08-31 CRAN (R 3.3.2)
>  munsell      0.4.3   2016-02-13 CRAN (R 3.3.2)
>  nlme         3.1-131 2017-02-06 CRAN (R 3.3.2)
>  plyr         1.8.4   2016-06-08 CRAN (R 3.3.2)
>  psych        1.6.12  2017-01-08 CRAN (R 3.3.2)
>  purrr      * 0.2.2   2016-06-18 CRAN (R 3.3.2)
>  R6           2.2.0   2016-10-05 CRAN (R 3.3.2)
>  Rcpp         0.12.9  2017-01-14 CRAN (R 3.3.2)
>  readr      * 1.0.0   2016-08-03 CRAN (R 3.3.3)
>  readxl       0.1.1   2016-03-28 CRAN (R 3.3.2)
>  reshape2     1.4.2   2016-10-22 CRAN (R 3.3.2)
>  rvest        0.3.2   2016-06-17 CRAN (R 3.3.2)
>  scales       0.4.1   2016-11-09 CRAN (R 3.3.2)
>  stringi      1.1.2   2016-10-01 CRAN (R 3.3.2)
>  stringr      1.2.0   2017-02-18 CRAN (R 3.3.2)
>  tibble     * 1.2     2016-08-26 CRAN (R 3.3.2)
>  tidyr      * 0.6.1   2017-01-10 CRAN (R 3.3.2)
>  tidyverse  * 1.1.1   2017-01-27 CRAN (R 3.3.2)
>  withr        1.0.2   2016-06-20 CRAN (R 3.3.3)
>  xml2         1.1.1   2017-01-24 CRAN (R 3.3.2)

**Comment by nickhepler** [comment]
> # Update
> Updated packages including readr; still get the same error.
> 
> # Session Info
> `devtools::session_info()`
> 
> Session info -------------------------------------------------------------------------------------
>  setting  value                       
>  version  R version 3.3.3 (2017-03-06)
>  system   x86_64, linux-gnu           
>  ui       RStudio (1.0.136)           
>  language (EN)                        
>  collate  en_US.UTF-8                 
>  tz       America/New_York            
>  date     2017-04-01                  
> 
> Packages -----------------------------------------------------------------------------------------
>  package    * version  date       source        
>  assertthat   0.1      2013-12-06 CRAN (R 3.3.2)
>  broom        0.4.2    2017-02-13 CRAN (R 3.3.2)
>  colorspace   1.3-2    2016-12-14 CRAN (R 3.3.2)
>  curl         2.4      2017-03-24 CRAN (R 3.3.3)
>  DBI          0.6-1    2017-04-01 CRAN (R 3.3.3)
>  devtools     1.12.0   2016-12-05 CRAN (R 3.3.3)
>  digest       0.6.12   2017-01-27 CRAN (R 3.3.2)
>  dplyr      * 0.5.0    2016-06-24 CRAN (R 3.3.2)
>  forcats      0.2.0    2017-01-23 CRAN (R 3.3.2)
>  foreign      0.8-67   2016-09-13 CRAN (R 3.3.1)
>  ggplot2    * 2.2.1    2016-12-30 CRAN (R 3.3.2)
>  gtable       0.2.0    2016-02-26 CRAN (R 3.3.2)
>  haven        1.0.0    2016-09-23 CRAN (R 3.3.2)
>  hms          0.3      2016-11-22 CRAN (R 3.3.2)
>  httr         1.2.1    2016-07-03 CRAN (R 3.3.2)
>  jsonlite     1.3      2017-02-28 CRAN (R 3.3.2)
>  lattice      0.20-34  2016-09-06 CRAN (R 3.3.1)
>  lazyeval     0.2.0    2016-06-12 CRAN (R 3.3.2)
>  lubridate    1.6.0    2016-09-13 CRAN (R 3.3.2)
>  magrittr     1.5      2014-11-22 CRAN (R 3.3.2)
>  memoise      1.0.0    2016-01-29 CRAN (R 3.3.3)
>  mnormt       1.5-5    2016-10-15 CRAN (R 3.3.2)
>  modelr       0.1.0    2016-08-31 CRAN (R 3.3.2)
>  munsell      0.4.3    2016-02-13 CRAN (R 3.3.2)
>  nlme         3.1-131  2017-02-06 CRAN (R 3.3.2)
>  plyr         1.8.4    2016-06-08 CRAN (R 3.3.2)
>  psych        1.7.3.21 2017-03-22 CRAN (R 3.3.3)
>  purrr      * 0.2.2    2016-06-18 CRAN (R 3.3.2)
>  R6           2.2.0    2016-10-05 CRAN (R 3.3.2)
>  Rcpp         0.12.10  2017-03-19 CRAN (R 3.3.3)
>  readr      * 1.1.0    2017-03-22 CRAN (R 3.3.3)
>  readxl       0.1.1    2016-03-28 CRAN (R 3.3.2)
>  reshape2     1.4.2    2016-10-22 CRAN (R 3.3.2)
>  rvest        0.3.2    2016-06-17 CRAN (R 3.3.2)
>  scales       0.4.1    2016-11-09 CRAN (R 3.3.2)
>  stringi      1.1.3    2017-03-21 CRAN (R 3.3.3)
>  stringr      1.2.0    2017-02-18 CRAN (R 3.3.2)
>  tibble     * 1.3.0    2017-04-01 CRAN (R 3.3.3)
>  tidyr      * 0.6.1    2017-01-10 CRAN (R 3.3.2)
>  tidyverse  * 1.1.1    2017-01-27 CRAN (R 3.3.2)
>  withr        1.0.2    2016-06-20 CRAN (R 3.3.3)
>  xml2         1.1.1    2017-01-24 CRAN (R 3.3.2)

**Comment by mrscholz** [comment]
Same problem on Win10, if I first load the `tidyverse` package. First loading `readr` and` tidyverse` afterwards avoids the error, however.

**Comment by hadley** [comment]
Something must be getting corrupted in your readr install — I can't understand why that file wouldn't be there.

Oh, unless your regular library isn't in `"~/R/x86_64-pc-linux-gnu-library/3.3"` and R is getting confused?

**Comment by pmoneill** [comment]
I have the same problem with read_csv on Win10 even after removing and reinstalling R, R Studio, and Tidyverse like nickhelper did in Linux.  read.csv and R Studio import work fine.

Error in date_names_lang(date_names) : 
  cannot open file '~/R/win-library/3.4/readr/R/sysdata.rdb': No such file or directory
In addition: Warning message:
In date_names_lang(date_names) : restarting interrupted promise evaluation

**Comment by ldecicco-USGS** [comment]
Checking in with the same problem, as well as a few users that reported it to me initially.

**Comment by mrscholz** [comment]
I have the same error on some of the packages included in the tidyverse. To avoid the problem, I found that it helps to load the packages that are affected **before** adding tidyverse to the current R session.

**Comment by hadley** [comment]
A full reprex would be useful.

**Comment by pmoneill** [comment]
In Windows 10 Pro, Version 1703, OS Build, 15063.413, on Dell XPS 8900
Start a new RStudio Version 1.0.143  session and load the tidyverse, which includes the readr library which includes the read_csv function.
Try to read any csv file and get error:
> library("tidyverse", lib.loc="~/R/win-library/3.4")
Loading tidyverse: ggplot2
Loading tidyverse: tibble
Loading tidyverse: tidyr
Loading tidyverse: readr
Loading tidyverse: purrr
Loading tidyverse: dplyr
Conflicts with tidy packages --------------------------------------------------
filter(): dplyr, stats
lag():    dplyr, stats
> Purchases <- read_csv("DailyPeakOffPeakPurchases.csv")
Error in date_names_lang(date_names) : 
  cannot open file '~/R/win-library/3.4/readr/R/sysdata.rdb': No such file or directory
In addition: Warning message:
In date_names_lang(date_names) : restarting interrupted promise evaluation
> 
However, starting a new RStudio session and loading readr separately (and redundantly?) BEFORE loading the tidyverse, as mrsholz suggests, works!
> library("readr", lib.loc="~/R/win-library/3.4")
> library("tidyverse", lib.loc="~/R/win-library/3.4")
Loading tidyverse: ggplot2
Loading tidyverse: tibble
Loading tidyverse: tidyr
Loading tidyverse: purrr
Loading tidyverse: dplyr
Conflicts with tidy packages --------------------------------------------------
filter(): dplyr, stats
lag():    dplyr, stats
> purchases <- read_csv("DailyPeakOffPeakPurchases.csv")
Parsed with column specification:
cols(
  Year = col_integer(),
  Month = col_integer(),
  Day = col_integer(),
  Off = col_double(),
  RawPeak = col_double(),
  Peak = col_double()
)

**Comment by hadley** [comment]
Why are you specifying `lib.loc="~/R/win-library/3.4"`? I suspect that's at the root of the problem.

**Comment by ldecicco-USGS** [comment]
In RStudio, if you use the checkboxes in the Package tab to load a package, it automatically tacks in the `lib.loc` argument with that location:

![image](https://user-images.githubusercontent.com/1105215/27593370-9ddd971e-5b1c-11e7-930e-66669c11ef48.png)

That has caused a few minor issues for me over the years, so I always tell people to load the libraries explicitly by typing in the console (or of course even better in their script).

**Comment by ldecicco-USGS** [comment]
I do think that `lib.loc` is causing this havoc. When I made the screenshot above (just clicking `tidyverse` checkbox), and then later tried to use a function that calls `readr::read_csv`, I got the error again (I had thought I solved the issue by re-installing `readr` yesterday).

My `.libPaths` is:
```
.libPaths()
[1] "C:/Users/ldecicco/Documents/R/win-library/3.4"
[2] "C:/Program Files/R/R-3.4.0patched/library"
```

I would think `~/R/win-library/3.4` would be fine, but seems like it would make more sense for RStudio to default to `lib.loc=NULL` (or just not specify anything).

More troubleshooting:
* I restarted RStudio, and did NOT open `tidyverse`, and the `read_csv` function worked. 
* I loaded `tidyverse` with the vanilla `library` function, and the `read_csv` function still worked
* Restarted RStudio, loaded `tidyverse` with the vanilla `library` function, and the `read_csv` function worked
* Restarted RStudio, loaded `tidyverse` via the checkbox in RStudio, and the `read_csv` (via `readr`, not `tidyverse`) gave the mentioned error:

```
 Error in date_names_lang(date_names) : 
  cannot open file '~/R/win-library/3.4/readr/R/sysdata.rdb': No such file or directory 
```

**Comment by hadley** [comment]
It's probably a weird bug in `library()`, but it seems like an easy fix would be for RStudio to not add `lib.loc` by default.

cc @jjallaire @kevinushey

**Comment by kevinushey** [comment]
I can confirm that using a full path rather than a `~`-prefixed path works around this issue. Normally, library paths when set through e.g. `.libPaths()` are normalized before they're cached and stored; I'm guessing that `library()` is making a similar assumption behind the scenes and things are going wrong when an aliased path is passed.

**Comment by kevinushey** [comment]
For what it's worth, RStudio specifies the library path explicitly because it's possible to have multiple copies of a package installed in different libraries, so we need to make sure to load the one the user has actually requested.

**Comment by hadley** [comment]
@kevinushey as well as fully normalising the path, I'd argue that you shouldn't supply `lib.loc` if the path is the same as `.libPaths()[[1]]`

**Comment by DevGri** [comment]
Because I got this error too I have uninstalled/reinstalled RStudio 1.0.153. I also cleared the library path "~/R/win-library/3.4". After that I reinstalled tidyverse. 

I am not really sure, but I believe that tidyverse now install a lot of packages (more than I remember). After installation I load the tidyverse package by setting the checkmark in RStudio.

I make a test with a simple example: 
read_csv("a, b, c\n1,2,3\n4,5,6")

I got the error message:
Error in date_names_lang(date_names) : 
  cannot open file '~/R/win-library/3.4/readr/R/sysdata.rdb': No such file or directory

Directory and file exists! The only problem could be the ~ sign.
(~ = "C:/Users/Helmut/Documents" for me).

My Workaround:
Uninstall and reinstall readr (tidyverse have not to be uninstalled)
No all works fine.

I worked on a windows 10 computer. I haven't changed the default behavior of lib.loc in RStudio.
I think the problem will be in the installation package of tidyverse, because the readr installation works (I checked that both version numbers are equal (1.1.1)

Salzburg, 11.9.2017

**Comment by alexiswl** [comment]
I came across this issue with R 3.5.0 running in Centos 6.9.  When I specified R_LIBS_USER in my ~/.Renviron to ~/R/%V/library, I got the 'no such file or directory error'. 

I also removed the R_LIBS_SITE environment variable and in ~/.Rprofile set the libpaths to just the R_LIBS_USER to ensure that I could install all packages on our HPC without sudo. I had done with this with previous R versions without issue.

I tried reinstalling readr, restarting sessions in both command line R and RStudio but nothing worked. I resolved this issue by using the absolute path for my R_LIBS_USER variable. i.e to /home/<username>/R/%V/library in my ~/.Renviron

It's as if the tilda is not pointing correctly to my home directory, yet all other packages are importing from this directory fine.

Alexis.
